data/transactions/logic/opcodeExplain.go

```diff
@@ -0,0 +1,277 @@
+// Copyright (C) 2019-2023 Algorand, Inc.
+// This file is part of go-algorand
+//
+// go-algorand is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// go-algorand is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with go-algorand.  If not, see <https://www.gnu.org/licenses/>.
+
+package logic
+
+import "github.com/algorand/go-algorand/data/basics"
+
+// debugStackExplain explains the effect of an opcode over the stack
+// with 2 integers: deletions and additions, representing pops and inserts.
+// An opcode may delete a few variables from stack, then add a few to stack.
+type debugStackExplain func(*EvalContext) (int, int)
+
+// AppStateOperationT stands for the operation enum to app state, should be one of create, write, read, delete.
+type AppStateOperationT uint64
```

Comment:
nit: this doesn't seem to match our existing naming convention. Why not drop the `T` at the end?

Reply:
Renamed to `AppStateOpEnum` to be consistent with `AppStateEnum`.